### PR TITLE
feat: add rarity threshold for loot notifs

### PR DESCRIPTION
**Pull request by iProdigy**
Follow-up to #425

### CODE REVIEW

File: src/main/java/dinkplugin/DinkPluginConfig.java [hunk 1]
```diff
@@ -866,13 +866,27 @@ default String lootItemAllowlist() {
         name = "Item Denylist",
         description = "Never fire notifications for these items, despite value settings.<br/>" +
             "Place one item name per line (case-insensitive; asterisks are wildcards)",
-        position = 38,
+        position = 37,
```
Comment (pajlada) [inline src/main/java/dinkplugin/DinkPluginConfig.java:869]: Let's double check that allow list is still above deny list with this change

File: src/main/java/dinkplugin/DinkPluginConfig.java [hunk 2]
```diff
@@ -866,13 +866,27 @@ default String lootItemAllowlist() {
...
     )
     default String lootItemDenylist() {
         return "";
     }
 
+    @ConfigItem(
+        keyName = "lootRarityThreshold",
+        name = "Rarity Override (1 in X)",
+        description = "Fires notifications for sufficiently rare drops, despite the 'Min Loot value' threshold.<br/>" +
+            "Corresponds to a 1 in X chance. For example, 100 notifies for items with 1% drop rate (or rarer).<br/>" +
```
Comment (pajlada) [inline src/main/java/dinkplugin/DinkPluginConfig.java:880]: the parenthesis can be removed here

**Review by pajlada — APPROVED** [review]

**Comment by pajlada** [comment]
I have not tested this in-game but it looks sane

File: src/main/java/dinkplugin/DinkPluginConfig.java (hunk 1 above)
Comment (iProdigy) [reply src/main/java/dinkplugin/DinkPluginConfig.java:869]: confirmed
image

**Comment by iProdigy** [comment]
Confirmed works in-game

**Pull request merged**